### ISSUE TITLE
building ior on edison using darshan 3.0 does not work

### ISSUE THREAD

**Issue by shanedsnyder** [darshan-hpc/darshan#178]
In GitLab by @shanedsnyder on Jan 14, 2016, 15:24

IOR attempts to build a dynamic executable by default which shouldn't be the case. For Darshan 2.3, the IOR build succeeds despite defaulting to dynamic executable, but using Darshan 3.0 we get build errors. Temporary workaround is to pass "-static" in the LDFLAGS.

We need to figure out:

- why IOR is attempting to build dynamic executables by default
- and, why the build fails when using Darshan 3.0

**Comment by shanedsnyder** [comment]
In GitLab by @carns on Jan 19, 2016, 14:08

As of January 19 2016 we can no longer recreate the problem in the first bullet (ior attempting to build dynamic executables by default), this appears to have been resolved by some system change beyond our control.

Leaving this ticket open, though; we should still test why a dynamic build (which we would have to explicitly force now) doesn't succeed.  When dynamic linking it is Ok if Darshan doesn't produce a log, but we need to avoid causing the build to fail.

**Comment by shanedsnyder** [comment]
In GitLab by @shanedsnyder on Jan 25, 2016, 11:46

It looks like the build no longer fails either, when providing '-dynamic' in the LDFLAGS. Today I tried building IOR on Edison when passing that linker flag, and the build succeeds with the resulting executable being dynamic as expected. There is no resulting Darshan instrumentation in the executable, which I believe is also the expected behavior. Closing this issue.
